sched/scheduler/stateful_scheduler_test.go

```diff
@@ -311,11 +321,14 @@ func Test_StatefulScheduler_JobRunsToCompletion(t *testing.T) {
 	jobId, _ := s.ScheduleJob(jobDef)
 
 	// add additional saga data
-	sagaLogMock.EXPECT().LogMessage(saga.MakeStartTaskMessage(jobId, taskId, nil))
-	sagaLogMock.EXPECT().LogMessage(TaskMessageMatcher{Type: &sagaStartTask, JobId: "job1", TaskId: "task1", Data: gomock.Any()}).MaxTimes(1)
-	endMessageMatcher := TaskMessageMatcher{JobId: jobId, TaskId: taskId, Data: gomock.Any()}
-	sagaLogMock.EXPECT().LogMessage(endMessageMatcher)
-	sagaLogMock.EXPECT().LogMessage(saga.MakeEndSagaMessage(jobId))
+	gomock.InOrder(
```

Comment:
I'm assuming the lack of InOrder() was causing the flakyness?  Were the errors from mock looking for them in a different order? (I'm just curious as to what was going on.)